src/main/java/io/supertokens/inmemorydb/queries/UserMetadataQueries.java

```diff
@@ -0,0 +1,91 @@
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.inmemorydb.queries;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.inmemorydb.ConnectionWithLocks;
+import io.supertokens.inmemorydb.Start;
+import io.supertokens.inmemorydb.config.Config;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.execute;
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.update;
+import static io.supertokens.inmemorydb.config.Config.getConfig;
+
+public class UserMetadataQueries {
+
+    public static String getQueryToCreateUserMetadataTable(Start start) {
+        String tableName = Config.getConfig(start).getUserMetadataTable();
+        // @formatter:off
+        return "CREATE TABLE IF NOT EXISTS " + tableName + " ("
+                + "user_id VARCHAR(128) NOT NULL,"
+                + "user_metadata TEXT," 
+                + "PRIMARY KEY(user_id)" + " );";
+        // @formatter:on
+
+    }
+
+    public static int deleteUserMetadata(Start start, String userId) throws SQLException, StorageQueryException {
+        String QUERY = "DELETE FROM " + getConfig(start).getUserMetadataTable() + " WHERE user_id = ?";
+
+        return update(start, QUERY.toString(), pst -> pst.setString(1, userId));
+    }
+
+    public static int setUserMetadata_Transaction(Start start, Connection con, String userId, JsonObject metadata)
+            throws SQLException, StorageQueryException {
+
+        String QUERY = "INSERT INTO " + getConfig(start).getUserMetadataTable()
+                + "(user_id, user_metadata) VALUES(?, ?) "
+                + "ON CONFLICT(user_id) DO UPDATE SET user_metadata=excluded.user_metadata;";
+
+        return update(con, QUERY, pst -> {
+            pst.setString(1, userId);
+            pst.setString(2, metadata.toString());
+        });
+    }
+
+    public static JsonObject getUserMetadata_Transaction(Start start, Connection con, String userId)
+            throws SQLException, StorageQueryException {
+        ((ConnectionWithLocks) con).lock(userId + getConfig(start).getUserMetadataTable());
+
+        String QUERY = "SELECT user_metadata FROM " + getConfig(start).getUserMetadataTable() + " WHERE user_id = ?";
+
+        return execute(con, QUERY, pst -> pst.setString(1, userId), result -> {
+            if (result.next()) {
+                JsonParser jp = new JsonParser();
+                return jp.parse(result.getString("user_metadata")).getAsJsonObject();
+            }
+            return null;
```

Comment:
I think you should return an empty JSON object here instead of checking for `null` in the core, since maybe in the future we might have to query this in other places in the core too, and at that time, we will have to remember to do null checks.

Reply:
I think returning null from the DB could be useful in the future (e.g.: if we decide the endpoint should return if this was the first time creating or if it was an update). Also, other places in the core could just query this through the recipe function instead of directly accessing the DB. I figured why throw this info away on the interface level.